src/commands/dart_symbol_map/upload.rs

```diff
@@ -0,0 +1,180 @@
+use crate::utils::chunks::{upload_chunked_objects, Assemblable, ChunkOptions, Chunked};
+use crate::utils::dif::DifFile;
+use symbolic::common::ByteView;
+use symbolic::common::DebugId;
+
+struct DartSymbolMapObject<'a> {
+    bytes: &'a [u8],
+    name: &'a str,
+    debug_id: DebugId,
+}
+
+impl<'a> AsRef<[u8]> for DartSymbolMapObject<'a> {
+    fn as_ref(&self) -> &[u8] {
+        self.bytes
+    }
+}
+
+impl<'a> Display for DartSymbolMapObject<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "dartsymbolmap {}", self.name)
+    }
+}
+
+impl<'a> Assemblable for DartSymbolMapObject<'a> {
+    fn name(&self) -> Cow<'_, str> {
+        Cow::Borrowed(self.name)
+    }
+
+    fn debug_id(&self) -> Option<DebugId> {
+        Some(self.debug_id)
+    }
+}
+
+#[derive(Args, Clone)]
+pub(crate) struct DartSymbolMapUploadArgs {
+    #[arg(short = 'o', long = "org")]
+    #[arg(help = "The organization ID or slug.")]
+    pub(super) org: Option<String>,
+
+    #[arg(short = 'p', long = "project")]
+    #[arg(help = "The project ID or slug.")]
+    pub(super) project: Option<String>,
+
+    #[arg(value_name = "MAPPING")]
+    #[arg(
+        help = "Path to the dartsymbolmap JSON file (e.g. dartsymbolmap.json). Must be a JSON array of strings with an even number of entries (pairs)."
+    )]
+    pub(super) mapping: String,
+
+    #[arg(value_name = "DEBUG_FILE")]
+    #[arg(
+        help = "Path to the corresponding debug file to extract the Debug ID from. The file must contain exactly one Debug ID."
+    )]
+    pub(super) debug_file: String,
+}
+
+pub(super) fn execute(args: DartSymbolMapUploadArgs) -> Result<()> {
+    let mapping_path = &args.mapping;
+    let debug_file_path = &args.debug_file;
+
+    // Extract Debug ID(s) from the provided debug file
+    let dif = DifFile::open_path(debug_file_path, None)?;
+    let mut ids: Vec<DebugId> = dif.ids().filter(|id| !id.is_nil()).collect();
+
+    // Ensure a single, unambiguous Debug ID
+    ids.sort();
+    ids.dedup();
+    match ids.len() {
+        0 => bail!(
+            "No debug identifier found in the provided debug file ({}). Ensure the file contains an embedded Debug ID.",
+            debug_file_path
+        ),
+        1 => {
+            let debug_id = ids.remove(0);
+
+            // Validate the dartsymbolmap JSON: must be a JSON array of strings with even length
+            let mapping_file_bytes = ByteView::open(mapping_path)
+                .with_context(|| format!("Failed to read mapping file at {mapping_path}"))?;
+            let mapping_entries: Vec<Cow<'_, str>> =
+                serde_json::from_slice(mapping_file_bytes.as_ref())
+                    .context("Invalid dartsymbolmap: expected a JSON array of strings")?;
+
+            if mapping_entries.len() % 2 != 0 {
+                bail!(
+                    "Invalid dartsymbolmap: expected an even number of entries (pairs), got {}",
```

Comment:
The "pairs" here is confusing IMO.
```suggestion
                    "Invalid dartsymbolmap: expected an even number of entries, got {}",
```